packages/text-editor/src/components/extension/validateContent.ts

```diff
@@ -0,0 +1,43 @@
+import { Editor, Extension } from '@tiptap/core'
+
+export interface ValidateContentOptions {
+  onEmpty?: (isEmpty: boolean) => void
+  onValidate?: (html: string) => void
+}
+
+export interface ValidateContentStorage {
+  isEmpty: boolean
+}
+
+const handleChange = (editor: Editor, options: ValidateContentOptions, storage: ValidateContentStorage): void => {
+  if (options.onEmpty !== undefined && options.onEmpty !== null) {
+    if (storage.isEmpty !== editor.isEmpty) {
+      storage.isEmpty = editor.isEmpty
```

Comment:
Should we update storage even when not `onEmpty` callback was passed?

Reply:
Updated